root/artist/ArtistIndex.js

```diff
@@ -0,0 +1,305 @@
+  +numberOfRevisions: number,
+  +otherIdentities: $ReadOnlyArray<ArtistT>,
+  +recordings: ?$ReadOnlyArray<RecordingT>,
+  +releaseGroups: ?$ReadOnlyArray<ReleaseGroupT>,
+  +showAll: boolean,
+  +showVarious: boolean,
+  +wikipediaExtract: WikipediaExtractT,
+|};
+
+const ArtistIndex = ({
+  $c,
+  ajaxFilterFormUrl,
+  artist,
+  eligibleForCleanup,
+  filterForm,
+  hasFilter,
+  includingAll,
+  includingVarious,
+  legalName,
+  legalNameAliases,
+  legalNameArtistAliases,
+  numberOfRevisions,
+  otherIdentities,
+  recordings,
+  releaseGroups,
+  showAll,
+  showVarious,
+  wikipediaExtract,
+}: Props) => (
+  <ArtistLayout entity={artist} page="index">
+    {eligibleForCleanup ? (
+      <p className="cleanup">
+        {l(`This artist has no relationships, recordings, releases or release
+            groups, and will be removed automatically in the next few days.
+            If this is not intended, please add more data to this artist.`)}
+      </p>
+    ) : null}
+
+    <Annotation
+      annotation={artist.latest_annotation}
+      collapse
+      entity={artist}
+      numberOfRevisions={numberOfRevisions}
+    />
+
+    {legalName ? (
+      <RelatedArtists
+        artists={[legalName]}
+        extra={legalNameArtistAliases}
+        title={l('Legal name')}
+      />
+    ) : (legalNameAliases && legalNameAliases.length) ? (
+      <p>
+        <strong>{addColonText(l('Legal name'))}</strong>
+        {' '}
+        {commaOnlyListText(legalNameAliases)}
+      </p>
+    ) : null}
+
+    {otherIdentities && otherIdentities.length ? (
+      <RelatedArtists
+        artists={otherIdentities}
+        title={l('Also performs as')}
+      />
+    ) : null}
+
+    <WikipediaExtract
+      cachedWikipediaExtract={wikipediaExtract || null}
+      entity={artist}
+    />
+
+    <h2 className="discography">{l('Discography')}</h2>
+
+    <Filter
+      ajaxFormUrl={ajaxFilterFormUrl}
+      initialFilterForm={filterForm}
+    />
+
+    {(releaseGroups && releaseGroups.length) ? (
+      <form action="/release_group/merge_queue" method="post">
+        {/* TODO: MBS-10155 */}
+        <ReleaseGroupList
+          checkboxes="add-to-merge"
+          groupByType
+          releaseGroups={releaseGroups}
+          showRatings
+        />
+        {$c.user_exists ? (
+          <div className="row">
+            <FormSubmit label={l('Merge release groups')} />
+          </div>
+        ) : null}
+      </form>
+    ) : null}
+
+    {(recordings && recordings.length) ? (
+      <RecordingList
+        checkboxes="add-to-merge"
+        recordings={recordings}
+        showRatings
+      />
+    ) : null}
+
+    <p>
+      {(recordings && recordings.length) ? (
+        l('This artist has no release groups, only standalone recordings.')
+
+      ) : ((!releaseGroups || releaseGroups.length === 0) && hasFilter) ? (
+        l('No release groups found.')
+
+      ) : (!showAll && !showVarious) ? (
```

Comment:
Don't think we can really use a case statement, since there's no one value to switch on, there's four. I updated it to use if statements instead, which might look a little better.

Reply:
That's probably the best we can do :) Thanks!